src/main/java/teammates/storage/sqlentity/FeedbackResponse.java

```diff
@@ -65,18 +68,83 @@ protected FeedbackResponse() {
     }
 
     public FeedbackResponse(
-            FeedbackQuestion feedbackQuestion, FeedbackQuestionType type, String giver,
-            Section giverSection, String receiver, Section receiverSection
+            FeedbackQuestion feedbackQuestion, String giver,
+            Section giverSection, String recipient, Section recipientSection
     ) {
         this.setId(UUID.randomUUID());
         this.setFeedbackQuestion(feedbackQuestion);
-        this.setFeedbackQuestionType(type);
         this.setGiver(giver);
         this.setGiverSection(giverSection);
-        this.setReceiver(receiver);
-        this.setReceiverSection(receiverSection);
+        this.setRecipient(recipient);
+        this.setRecipientSection(recipientSection);
+    }
+
+    /**
+     * Creates a feedback response according to its {@code FeedbackQuestionType}.
+     */
+    public static FeedbackResponse makeResponse(
+            FeedbackQuestion feedbackQuestion, String giver,
+            Section giverSection, String receiver, Section receiverSection,
+            FeedbackResponseDetails responseDetails
+    ) {
+        FeedbackResponse feedbackResponse = null;
+        switch (responseDetails.getQuestionType()) {
```

Comment:
Please also change to `recipient` for this method.

Reply:
changed!